fission/src/test/scene/GizmoSceneObject.test.ts

```diff
@@ -0,0 +1,280 @@
+        attach: vi.fn(),
+        detach: vi.fn(),
+        setSize: vi.fn(),
+        addEventListener: vi.fn(),
+        removeEventListener: vi.fn(),
+        dragging: false,
+        enabled: true,
+        translationSnap: null,
+        rotationSnap: null,
+        setScaleSnap: vi.fn(),
+        axis: "XYZ",
+        mode: "translate",
+        object: null,
+    })),
+}))
+
+describe("GizmoSceneObject", () => {
+    let gizmoSceneObject: GizmoSceneObject
+    let mockMesh: THREE.Mesh
+    let mockParentObject: MirabufSceneObject
+
+    beforeEach(() => {
+        vi.clearAllMocks()
+
+        // Create mock mesh
+        mockMesh = new THREE.Mesh(new THREE.BoxGeometry(1, 1, 1), new THREE.MeshBasicMaterial({ color: 0x00ff00 }))
+
+        // Create mock parent object
+        mockParentObject = {
+            id: "test-parent-id",
+            mirabufInstance: {
+                parser: {
+                    rigidNodes: [{ id: "node1" as RigidNodeId }, { id: "node2" as RigidNodeId }],
+                },
+            },
+            mechanism: {
+                getBodyByNodeId: vi.fn(() => "mock-body-id"),
+            },
+            disablePhysics: vi.fn(),
+            enablePhysics: vi.fn(),
+            updateMeshTransforms: vi.fn(),
+        } as unknown as MirabufSceneObject
+    })
+
+    afterEach(() => {
+        if (gizmoSceneObject) {
+            gizmoSceneObject.dispose()
+        }
+    })
+
+    describe("setTransform()", () => {
+        beforeEach(() => {
+            gizmoSceneObject = new GizmoSceneObject("translate", 1.0, mockMesh, mockParentObject)
+        })
+
+        test("should apply translation transformation correctly", () => {
+            const targetPosition = new THREE.Vector3(5, 10, -3)
+            const transform = new THREE.Matrix4().makeTranslation(targetPosition.x, targetPosition.y, targetPosition.z)
+
+            gizmoSceneObject.setTransform(transform)
+
+            // Test that position is applied correctly
+            expect(gizmoSceneObject.obj.position.x).toBeCloseTo(targetPosition.x)
+            expect(gizmoSceneObject.obj.position.y).toBeCloseTo(targetPosition.y)
+            expect(gizmoSceneObject.obj.position.z).toBeCloseTo(targetPosition.z)
+
+            // Test that the position vector matches
+            expect(gizmoSceneObject.obj.position.equals(targetPosition)).toBe(true)
+        })
+
+        test("should apply rotation transformation correctly", () => {
+            const rotationAngle = Math.PI / 4 // 45 degrees
+            const rotationAxis = new THREE.Vector3(0, 1, 0) // Y-axis
+            const transform = new THREE.Matrix4().makeRotationAxis(rotationAxis, rotationAngle)
+
+            gizmoSceneObject.setTransform(transform)
+
+            // Test that rotation is applied correctly
+            expect(gizmoSceneObject.obj.rotation.y).toBeCloseTo(rotationAngle)
+            expect(Math.abs(gizmoSceneObject.obj.rotation.x)).toBeLessThan(0.001)
```

Comment:
What is with the inconsistent use of .toBeCloseTo vs .toBeLessthan for checking that things are almost exactly the same?